transport/httptransport/http_transport.go

```diff
@@ -248,50 +249,54 @@ func (t *transfer) execute(ctx context.Context) error {
 		// start the http transfer
 		remaining := t.dealInfo.DealSize - t.nBytesReceived
 		reqErr := t.doHttp(ctx, req, of, remaining)
-		if reqErr != nil {
-			t.dl.Infow(duuid, "one http req done", "http code", reqErr.code, "outputErr", reqErr.Error())
-		} else {
-			t.dl.Infow(duuid, "http req finished with no error")
-		}
-
 		if reqErr == nil {
-			t.dl.Infow(duuid, "http req done without any errors")
+			t.dl.Infow(duuid, "http request completed successfully")
 			// if there's no error, transfer was successful
 			break
 		}
+
+		t.dl.Infow(duuid, "http request error", "http code", reqErr.code, "outputErr", reqErr.Error())
+
 		_ = of.Close()
 
 		// check if the error is a 4xx error, meaning there is a problem with
 		// the request (eg 401 Unauthorized)
 		if reqErr.code/100 == 4 {
-			msg := fmt.Sprintf("terminating http req: received %d response from server", reqErr.code)
+			msg := fmt.Sprintf("terminating http request: received %d response from server", reqErr.code)
 			t.dl.LogError(duuid, msg, reqErr)
 			return reqErr.error
 		}
 
 		// do not resume transfer if context has been cancelled or if the context deadline has exceeded
 		err = reqErr.error
 		if xerrors.Is(err, context.Canceled) || xerrors.Is(err, context.DeadlineExceeded) {
-			t.dl.LogError(duuid, "terminating http req as context cancelled or deadline exceeded", err)
-			return fmt.Errorf("transfer context err: %w", err)
+			t.dl.LogError(duuid, "terminating http request: context cancelled or deadline exceeded", err)
+			return fmt.Errorf("transfer context canceled err: %w", err)
+		}
+
+		// If some data was transferred, reset the back-off count to zero
+		if t.nBytesReceived > st.Size() {
```

Comment:
where does this reset happen ? We only log below

Reply:
Fixed